packages/@lwc/engine-core/src/framework/def.ts

```diff
@@ -131,6 +132,17 @@ function createComponentDef(Ctor: LightningElementConstructor): ComponentDef {
             );
         }
 
+        if (
+            isReportingEnabled() &&
+            (ctorShadowSupportMode === ShadowSupportMode.Any ||
+                ctorShadowSupportMode === ShadowSupportMode.Native)
+        ) {
+            report(ReportingEventId.ShadowSupportModeUsage, {
+                tagName: ctorName,
```

Comment:
Using tagName here to avoid having to make changes to our o11y schema in 248/patch. This should use its own field 'ctorName' or 'name' rather than 'tagName'.